File: frontend/types/notification.ts
export interface Notification {
  id: string;
  user_id: string;
  type: string;
  title: string;
  message: string;
  data: Record<string, unknown>;
  severity: "info" | "warning" | "error" | "success";
  is_seen: boolean;
  created_at: string;
}

export interface NotificationListResponse {
  items: Notification[];
  unread_count: number;
}

export interface InsightCard {
  id: string;
  type: string;
  title: string;
  body: string;
  severity: "info" | "warning" | "error" | "success";
  icon: string;
}

File: backend/app/schemas/notification.py
import uuid
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class NotificationSchema(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    user_id: uuid.UUID
    type: str
    title: str
    message: str
    data: dict
    severity: str
    is_seen: bool
    created_at: datetime


class NotificationListResponse(BaseModel):
    items: list[NotificationSchema]
    unread_count: int
